src/NBitcoin/Block.cs

```diff
@@ -95,6 +123,10 @@ internal void SetNull()
 
         public void ReadWrite(BitcoinStream stream)
         {
+            // Propagate legacy static flags.
+            if (!stream.Serializing)
```

Comment:
Yes, one way would be to have another flag/class but that would add extra arguments to methods that already have TransactionOptions/ProtocolVersion. It would be like doing similar work twice or in two different ways.